diff --git a/third_party/build/bin/freediameter_build.sh b/third_party/build/bin/freediameter_build.sh
index b67b325be..3cb3753d1 100755
--- a/third_party/build/bin/freediameter_build.sh
+++ b/third_party/build/bin/freediameter_build.sh
@@ -16,7 +16,7 @@ SCRIPT_DIR="$(dirname "$(realpath "$0")")"
 source "${SCRIPT_DIR}"/../lib/util.sh
 GIT_URL=https://github.com/lionelgo/opencord.org.freeDiameter.git
 GIT_COMMIT=13b0e7de0d66906d50e074a339f890d6e59813ad
-PKGVERSION=0.0.1
+PKGVERSION=1.2.0
 ITERATION=1
 VERSION="${PKGVERSION}"-"${ITERATION}"
 PKGNAME=oai-freediameter
